### DISCUSSION TITLE
Cannot call `raise_for_status` as the request instance has not been set on this response.

### DISCUSSION THREAD

**Discussion by DeoLeung** [encode/httpx#3252] [Potential Issue] [answered]
if upstream failed with 599, calling `raise_for_status` raises `RuntimeError`, is it a bug?

**Comment by lovelydinosaur** [comment]
Could you include a runnable example to demonstrate?

**Reply by DeoLeung** [reply]
argh, I made a pytest and it's passing :)

should be something wrong with my subclass client, I will double check it

**Comment by DeoLeung** [comment, marked as answer]
the problem was, I did some other error handling logic in a subclass client, which return a new `httpx.Response`
```python
def handle_error(res: httpx.Response):
  return httpx.Response(...)
```
now doing below works
```python
def handle_error(res: httpx.Response):
  return httpx.Response(..., request=res.request)
```
